### ISSUE TITLE
GitHubの練習をする - clone, commit, push, pull

### ISSUE THREAD

**Issue by doish** [doish/contact#20]
作業時間対象?: はい
このissueをいつやるか？: mydsl-phpのソースコードを実際更新する前までに、いつでも

GitHubの操作はmydsl-phpプロジェクトで必須になる技能です。
また、Gitが操作できますと人に言えると十分なアピールになります。
（Gitは技術職でない人はまず使えませんし、技術職の人でも古い技術しかさわってない人は使えないので差別化できることを覚えておきましょう。現場ではすでに十分普及しています。開発なら使うと思っておきましょう。）

https://github.com/doish/training.git
トレーニング用のリポジトリ（＝プロジェクトにひもづく、Git上のソースコードの一式）を用意したのでこちらを好きにコミットしてください。
トレーニング用なので中身はあってないようなものです。
ただし、最後までやるにはtrainingリポジトリへのユーザ登録の完了が必要で、
それは@doishからの招待(invite)メールをみなさんが承諾(accept)する必要があります。（別途送ります。）

用語の説明
- git clone
まだ自分のPC（ローカル）にソースコードがない時に、gitからソースコード一式を取ってきます。（もうすでにある時はgit pull。最後に説明）
この画面のあとは適当にNextとかFinishとかを押してください。Open Projectも押してください。
Apache NetBeans上だとTeam > Remote > Clone
Repository URL はプロジェクトごとに違いますが今回は上記のhttps://github.com/doish/training.git
UserとPasswordは自分のGitHubのIDのものです。
![image](https://user-images.githubusercontent.com/17733955/58393495-f1f0d080-8079-11e9-9fa9-f88470862ed1.png)

- git commit
ソースコードに新しい行を加えると下のように緑のマークが表示されます。（行を修正した場合は青のマーク）
![image](https://user-images.githubusercontent.com/17733955/58393654-80655200-807a-11e9-9098-50c407f0b92c.png)
この変更を、ローカルのgitに保存するのがgit commitです。
（この時点だとネットワークのgit(github)にはまだ反映されていません。）
プロジェクトを右クリックしてGit > Commitでコミット画面に移ります。
コミットメッセージを入力して（自由）コミットボタンを押してください。
![image](https://user-images.githubusercontent.com/17733955/58393738-f49ff580-807a-11e9-995f-d66e30587043.png)

- git push
ローカルのgitに保存した変更を、ネットワークのgitに反映するのがgit pushです。
コミット後、プロジェクトを右クリック Git > Remote > Push
Next > Next > Finishで完了です。
メッセージは出ないのでhttps://github.com/doish/training/ にアクセスして、自分の変更が反映されていることを確認してください。
![image](https://user-images.githubusercontent.com/17733955/58394121-a986e200-807c-11e9-8815-b2631bee6756.png)

- git pull
誰かがもし https://github.com/doish/training.git を変更したならば、あなたのローカルのgitにそれを取り込むことができます。それがgit push です。
プロジェクトを右クリック > Git > Remote > Pull > Next > Finish
（画像なし）
これが終わっていないと、あなたが新しい変更をローカルのgitに加えても、ネットワークのgitにpushすることはできません。

基本の流れは以上です。何度も言いますがhttps://github.com/doish/training.git は練習用のリポジトリなので、色々試してもらって結構です。

チーム開発という意味だとこれだけだと不十分で、「ブランチ管理」「プルリクエスト」がかかわってきますが、別途説明します。一人で開発する時はこのissueに書いてあることで十分GitHubが使えています。

何か質問があればこちらまで。

- git と githubの違いについて。
gitはソースコード管理の仕組みです。GitHubはgitの仕組みを提供するWebサービスです。gitが使えるWebサービスは他にGitLab等もあります。

**Comment by NanamiShinya** [comment]
お疲れ様です
git cloneの段階で止まってしまいました…
これは既にソースコードが入っているということでしょうか…？

![スクリーンショット 2019-05-28 21 39 32](https://user-images.githubusercontent.com/50830466/58479409-34f29700-8193-11e9-9b0e-f38e72f0c4b2.jpg)

**Comment by doish** [comment]
はい。@NanamiShinya さんはこの間対面でここの練習をして、trainingフォルダがすでにあります。ただし、そのあと@doish が https://github.com/doish/training.git の中身をGitHubアカウント名の関係で作り直したので、別物になっています。なのでgit pullもできないはずです。
一度 `/Users/Nanami/NetBeans` にあるtrainingフォルダを丸ごと削除してからgit cloneしてください。
以降は、git pullです。

**Comment by doish** [comment]
NetBeans上でGit Userの登録をしましょう。
```
[user]
	useConfigOnly = true
	name = あなたのGitHub username
	email = GitHubに登録したときのメールアドレス
[filter "lfs"]
	smudge = git-lfs smudge -- %f
	process = git-lfs filter-process
	required = true
	clean = git-lfs clean -- %f
```

Team > Repository(Git) > Open Global Configulation
